src/frontend/src/handler/alter_streaming_rate_limit.rs

```diff
@@ -56,6 +58,17 @@ pub async fn handle_alter_streaming_rate_limit(
             let reader = session.env().catalog_reader().read_guard();
             let (source, schema_name) =
                 reader.get_source_by_name(db_name, schema_path, &real_table_name)?;
+            if let Some(prev_limit) = source.rate_limit {
+                if rate_limit == SOURCE_RATE_LIMIT_PAUSED
+                    || (prev_limit != 0 && rate_limit == SOURCE_RATE_LIMIT_RESUMED)
+                {
+                    return Err(InvalidInputSyntax(
+                        "PAUSE or RESUME is invalid when the stream has pre configured ratelimit."
```

Comment:
"PAUSE or RESUME is invalid when the stream has pre configured ratelimit. Just use `SET source_rate_limit = 0`"